.gitignore

```diff
@@ -12,11 +12,11 @@ tmp/
 
 # gradle build
 .gradle
+.out/
 build
 dependencies.lock
 **/dependencies.lock
 gradle/
-out
```

Comment:
We can keep the `out` entry if anybody feels strongly about it.

It stems from this rather old PR and is definitely supposed to be grouped with the `gradle build` section (as there were far fewer sections then): https://github.com/apache/iceberg/commit/c59138eadcfb4eb2558a15f73c00b54528d02e12

My git on my work computer (git version 2.30.1 (Apple Git-130)) does not ignore the out folders with this entry, but it does with the way I've updated it.